rand_distr/src/poisson.rs

```diff
@@ -69,6 +72,9 @@ where F: Float + FloatConst, Standard: Distribution<F>
         if !(lambda > F::zero()) {
             return Err(Error::ShapeTooSmall);
         }
+        if lambda.is_infinite() {
+            return Err(Error::ParameterInfinite);
+        }
```

Comment:
I think it would be better to test that the value is finite first, so that NaN returns `Error::NonFinite`.
```suggestion
        if !lambda.is_finite() {
            return Err(Error::NonFinite);
        }
        if !(lambda > F::zero()) {
            return Err(Error::ShapeTooSmall);
        }
```

Reply:
Good point, I've changed it